include/GoogleSync/GoogleSync.php

```diff
@@ -46,1088 +53,249 @@
-         * So we check to make sure it has an ID to determine Success/Failure.
-         */
-        if (isset($greturn->id)) {
-            return $this->setLastSync($event, $event_remote->getId());
-        } else {
-            return false;
         }
+        //else
+        throw new GoogleSyncException('Something went wrong with the requested action');
     }
 
     /**
-     * Delete SuiteCRM Meeting
-     *
-     * @param Meeting $meeting SuiteCRM Meeting Bean
-     *
-     * @return bool Success/Failure of setLastSync, since that's what saves the record
-     */
-    public function delMeeting($meeting)
-    {
-        $meeting->deleted = '1';
-        $meeting->gsync_id = '';
-        return $this->setLastSync($meeting);
-    }
-
-    /**
-     * Delete Google Event
+     * Perform the sync for a user
      *
-     * @param \Google_Service_Calendar_Event $event \Google_Service_Calendar_Event Object
-     * @param String $meeting_id SuiteCRM Meeting Id
+     * @param string $id The SuiteCRM user id
      *
      * @return bool Success/Failure
+     * @throws Exception Rethrows if caught from GoogleSyncBase::initUserService
      */
-    public function delEvent($event, $meeting_id)
+    public function doSync($id)
     {
-        // Make sure the calendar service is set up
-        if (!isset($this->gService)) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The Google Service is not set up. See setGService Method.');
-            return false;
-        }
-
-        // Make sure we got a meeting_id
-        if (!isset($meeting_id)) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'This function requires a meeting id as the 2nd parameter');
-            return false;
-        }
-
-        $return = $this->gService->events->delete($this->calendarId, $event->getId());
-
-        // Pull the status code returned to determine Success/Failure
-        $statusCode = $return->getStatusCode();
-
-        if ($statusCode >= 200 && $statusCode <= 299) {
-         // 2xx statusCode = success
-            $GLOBALS['log']->debug(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Received Success Status Code: ' . $statusCode . ' on delete.');
-
-            // This removes the gsync_id reference from the table.
-            $sql = "UPDATE Meetings SET gsync_id = '' WHERE id = '" . $meeting_id . "'";
-            $res = $this->db->query($sql);
-
-            $this->syncedList[] = $meeting_id;
-            return true;
-        } else {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Received Failure Status Code: ' . $statusCode . ' on delete!');
-            return false;
+        try {
+            $this->initUserService($id);
+        } catch (Exception $e) {
+            // TODO: handle the exception instead just loggin it, it will be logged if unhandled by suitecrm core.
```

Comment:
@gymad 
All the Exceptions caught/thrown here are handled in syncAllUsers(). I suppose this log output is redundant.

Reply:
removed redundant logging.